controlpanel/api/rules.py

```diff
@@ -27,6 +27,9 @@ def is_app_admin(user, obj):
     :param obj App|AppS3Bucket|UserApp: The object to check
     """
     if obj is None:
+        return False  # TODO: Check this change is appropriate.
```

Comment:
I think this makes sense as it's more conservative and assumes someone is not an app admin when the "app" is `None` 👍 

I wonder if this condition could be `if not obj:`? `obj is None` seems very specific. 
As you said it's a bit of an edge case but you never know with these things. For "sure" returning `False` is a better default.

Reply:
Thanks for this, you raise some good points about Alice, Bob, and Eve. I think that as long as we have a record of who added whom and when (which we presumably do), I am content for admins to be able to create other admins. This is because:
 - I think the risk of Eve getting added accidentally is probably not that large
 - As it stands there is a burden of admin on superusers for exactly this reason
 - It's particularly a pain when someone has left the organisation without handing over admin duties (because they couldn't do so)
 - The superusers are further from projects and so less able to judge the suitability of Eve as an app admin, and more likely to just grant her the role in exchange for not seeming to be a blocker or jobsworth

Reply:
@samtazzyman ack on the points you make. For me it was more a case of everyone just elevating everyone else to the role of app admin because it's just more convenient to work with those sets of privileges (in which case, why have app admins). However, from you say, I guess this probably won't happen given the current practices in the org.

Reply:
@ntoll bear in mind something that you probably already know from previous comment, someone is an app admin for a **specific** app.

Being an "app admin" is more of a state (as in there is a `UserApp` record with `admin=True`) rather than a "role" as in the traditional sense as in "someone is an app admin and therefore can admin any app". 

Hence make someone an app admin out of laziness wouldn't have an app-specific blast radius.